backend/gateway/mux/mux.go

```diff
@@ -275,6 +275,16 @@ func New(unaryInterceptors []grpc.UnaryServerInterceptor, assets http.FileSystem
 		httpMux.HandleFunc("/debug/pprof/", pprof.Index)
 	}
 
+	_, ok := gatewayCfg.Stats.Reporter.(*gatewayv1.Stats_PrometheusReporter_)
+	if ok && metricsHandler != nil {
+		metricsPath := "/metrics"
+		promCfg := gatewayCfg.Stats.GetPrometheusReporter()
+		if promCfg.HandlerPath != "" {
+			metricsPath = promCfg.HandlerPath
+		}
+		httpMux.Handle(metricsPath, metricsHandler)
+	}
```

Comment:
How do you want to handle the error case here? 

If we are expecting a metrics handler to be configured but its the incorrect type will fail silently, I think will want to bail out here and fail to start up the application, instead of finding out at runtime the application is not emitting metrics.

Reply:
Changed to return an error when metrics handler isn't nil and config isn't prometheus